controller/src/main/java/io/pravega/controller/server/SegmentHelper.java

```diff
@@ -88,64 +103,77 @@ public SegmentHelper(final ConnectionFactory clientCF, HostControllerStore hostS
         return Controller.NodeUri.newBuilder().setEndpoint(host.getIpAddr()).setPort(host.getPort()).build();
     }
 
-    public CompletableFuture<Boolean> createSegment(final String scope,
-                                                    final String stream,
-                                                    final long segmentId,
-                                                    final ScalingPolicy policy,
-                                                    String controllerToken,
-                                                    final long clientRequestId) {
-        final CompletableFuture<Boolean> result = new CompletableFuture<>();
-        final String qualifiedStreamSegmentName = getQualifiedStreamSegmentName(scope, stream, segmentId);
-        final Controller.NodeUri uri = getSegmentUri(scope, stream, segmentId);
-        final WireCommandType type = WireCommandType.CREATE_SEGMENT;
-        final long requestId = (clientRequestId == RequestTag.NON_EXISTENT_ID) ? idGenerator.get() : clientRequestId;
-
-        final FailingReplyProcessor replyProcessor = new FailingReplyProcessor() {
-            @Override
-            public void connectionDropped() {
-                log.warn(requestId, "CreateSegment {} Connection dropped", qualifiedStreamSegmentName);
-                result.completeExceptionally(
-                        new WireCommandFailedException(type, WireCommandFailedException.Reason.ConnectionDropped));
+    RawClient getConnection(PravegaNodeUri nodeUri) {
+        synchronized (lock) {
+            if (client == null || client.isClosed()) {
+                client = new RawClient(nodeUri, connectionFactory);
```

Comment:
do we multiplex all requests on the same client?

Reply:
as per my understanding, that's one of the benefits to use RawClient; in recent commit, the 'requestId' has been changed to use `connection.getFlow().getNextSequenceNumber`.  @shrids could you please give more explanation for the question?

Reply:
@huide9 the question more related to the performance aspect. 
```
Before this PR the SegmentHelper used  `SegmentStoreConnectionManager` to manage 
connections with SegmentStore. This maintains a cache of connection pools to every 
SegmentStore where a pool could contain 
`SegmentStoreConnectionManager#MAX_CONCURRENT_CONNECTIONS` (500) network connections
per Segment store.
```

The current code is replacing this with a single RawClient per segment store. The question is do we need to create a pool of RawClient per segment store. 
As per my current understanding this is not required since having a pool of RawClients per Segment store would also translate into using same underlying network connection.

cc: / @shiveshr @tkaitchuck

Reply:
The question is more about -- a client is specific to segment store instance.. right?
shouldnt we maintain a map of nodeUri to client. and return an existing client for that node uri. 
here it appears we could get back a client which could be for a different pravegaNodeUri.

Reply:
shall we put this RawClient modification in a separate PR, so the current one (SegmentHelper related) can go ahead - I can go ahead file an issue to track this, please let me know. thanks